Add provider model for ARM processor

rynowak opened the pull request:
This change adds a 'provider' model for our ARM processor so we can
customize the handling of different resource families. This is a mirror
of how the actual deployment engine is implemented and will make it easier for
us to extend.

This also adds support for nested deployments (modules) in the 'local' execution model of the deployment client. We didn't really have any good tests for this code path so I added some pretty expansive new integration tests.

rynowak committed:
Add provider model for ARM processor

This change adds a 'provider' model for our ARM processor so we can
customize the handling of different resource families. This is a mirror
of how the deployment engine is implemented and will make it easier for
us to extend.

This also adds support for nested deployments (modules).

rynowak committed:
Add preliminary 'builder' support (#1594)

Fixes: #1556

This change adds preliminary support for builds as part of 'rad.yaml'.
Users can describe build tasks as part of 'rad.yaml' and those build
tasks will run as part of deployment. In particular the build results
can be consumed by IAC code as a parameter.

This change is preliminary and adds the general framework along with a
**very** simple builder integration for docker/Dockerfiles. The
intention here is that we will grow a plugin ecosystem of 'builders'
that can be maintained by the community and is loosely coupled to
Radius. We need build support for inner-loop experiences that work with
`rad`.

asilverman committed:
comment out test that is causing flakiness in ci build (#1663)

* use a standard sku to avoid quota issues

* ensure random name always starts with a letter

Co-authored-by: asilverman <asilverman@microsoft.com>

asilverman committed:
delete resources in parallel (#1573)

Co-authored-by: asilverman <asilverman@microsoft.com>

dependabot[bot] committed:
Bump actions/github-script from 4 to 5 (#1645)

tcnghia reviewed (commented):
My first batch of comment, more coming

asilverman committed:
separate command from output (#1666)

Co-authored-by: asilverman <asilverman@microsoft.com>

dependabot[bot] committed:
Bump helm/kind-action from 1.0.0 to 1.2.0 (#1646)

asilverman committed:
add nil check for default choice (#1679)

* add nil check for default choice

* improve the grammar on the prompt for consistency

Co-authored-by: asilverman <asilverman@microsoft.com>

rynowak committed:
Bugfixes for API Service (#1674)

- Operation URLs should include the path prefix of our APIService
- Deserialization was using wrong type/struct
- PUT operations should 'scrape' secrets to avoid persisting them in
  resource bodies in plaintext.
- List/GET operations are not including the name field for resources

jkotalik reviewed (approved):
Great tests

tcnghia reviewed (approved):
The rest are test files so I only have the original comments.

jkotalik committed:
Allows running apiserver without TLS (#1672)

* Allows running apiserver without tls

* Allow for override and add documentation

* Delete some outdated docs

* Cleanup comment

* Use env var

Commit:
Update CODEOWNERS (#1688)

Commit:
Update GH address (#1686)

Commit:
Revert "Bump actions/github-script from 4 to 5 (#1645)" (#1687)

This reverts commit 579ee1358a3e4c100c69cfc38a9344495571c7e9.

youngbupark committed:
Rename github.com/azure to github.com/project-radius in .go (#1694)

* Rename github.com/azure to github.com/project-radius

* Updating generate code

* rename url in Project

Co-authored-by: Justin Kotalik <jukotali@microsoft.com>

jkotalik committed:
Azure to project-radius (#1685)
